skimage/measure/tests/test_marching_cubes.py

```diff
@@ -74,5 +97,58 @@ def test_correct_mesh_orientation():
     np.testing.assert_array_equal(expected, corrected_faces1)
 
 
+def test_both_algs_same_result_ellipse():
+    # Performing this test on data that does not have ambiguities
+    
+    sphere_small = ellipsoid(1, 1, 1, levelset=True)
+    
+    vertices1, faces1, *_ = marching_cubes(sphere_small, 0)
+    vertices2, faces2, *_ = marching_cubes_lewiner(sphere_small, 0, allow_degenerate=False)
+    vertices3, faces3, *_ = marching_cubes_lewiner(sphere_small, 0, allow_degenerate=False, use_classic=True)
+    
+    # Order is different, best we can do is test equal shape and same vertices present
+    assert _same_mesh(vertices1, faces1, vertices2, faces2)
+    assert _same_mesh(vertices1, faces1, vertices3, faces3)
+
+
+def _same_mesh(vertices1, faces1, vertices2, faces2):
```

Comment:
Why the integer rounding scheme, and not a floating point comparison with tolerance?